MSBuildFileLogger Verbosity does not accept Verbosity.Verbose

```csharp
Task("Build")
    .IsDependentOn("Restore-NuGet-Packages")
    .Does(() =>
{
    MSBuild(solution, settings =>
        settings.SetConfiguration(configuration)
            .WithProperty("TreatWarningsAsErrors", "True")
            .SetVerbosity(Verbosity.Minimal)
            .AddFileLogger(new MSBuildFileLogger
            {
                Verbosity = Verbosity.Verbose
            }));
});
```

results in

```
========================================
Build
========================================
Executing task: Build
Microsoft (R) Build Engine version 15.1.1012.6693
Copyright (C) Microsoft Corporation. All rights reserved.

MSBUILD : Logger error MSB4103: "Verbose" is not a valid logger verbosity level.
An error occurred when executing task 'Build'.
Error: MSBuild: Process returned an error (exit code 1).
```

However, `Verbosity.Diagnostic` works great!

--- a/src/Cake.Common.Tests/Unit/Tools/MSBuild/MSBuildRunnerTests.cs
+++ b/src/Cake.Common.Tests/Unit/Tools/MSBuild/MSBuildRunnerTests.cs
@@ -869,7 +869,7 @@ namespace Cake.Common.Tests.Unit.Tools.MSBuild
                 // When
                 var result = fixture.Run();
                 // Then
-                Assert.Equal(@"/v:normal /target:Build /fl /flp:logfile=""C:/Working/A"";Encoding=E;Verbosity=Diagnostic /fl1 /flp1:Append;PerformanceSummary;NoSummary;ErrorsOnly;NoItemAndPropertyList;ShowCommandLine;ShowTimestamp;ShowEventId;Verbosity=Minimal /fl2 /flp2:WarningsOnly;Verbosity=Normal /fl3 /flp3:Verbosity=Quiet /fl4 /flp4:Verbosity=Verbose /fl5 /fl6 ""C:/Working/src/Solution.sln""", result.Args);
+                Assert.Equal(@"/v:normal /target:Build /fl /flp:logfile=""C:/Working/A"";Encoding=E;Verbosity=diagnostic /fl1 /flp1:Append;PerformanceSummary;NoSummary;ErrorsOnly;NoItemAndPropertyList;ShowCommandLine;ShowTimestamp;ShowEventId;Verbosity=minimal /fl2 /flp2:WarningsOnly;Verbosity=normal /fl3 /flp3:Verbosity=quiet /fl4 /flp4:Verbosity=detailed /fl5 /fl6 ""C:/Working/src/Solution.sln""", result.Args);
             }
 
             [Fact]

--- a/src/Cake.Common/Tools/MSBuild/MSBuildFileLogger.cs
+++ b/src/Cake.Common/Tools/MSBuild/MSBuildFileLogger.cs
@@ -98,7 +98,7 @@ namespace Cake.Common.Tools.MSBuild
             parameters.Add(ShowCommandLine ? "ShowCommandLine" : null);
             parameters.Add(ShowTimestamp ? "ShowTimestamp" : null);
             parameters.Add(ShowEventId ? "ShowEventId" : null);
-            parameters.Add(Verbosity != null ? $"Verbosity={Verbosity.Value.ToString()}" : null);
+            parameters.Add(Verbosity != null ? $"Verbosity={Verbosity.Value.GetMSBuildVerbosityName()}" : null);
 
             return string.Join(";", parameters.Where(p => p != null));
         }

--- a/src/Cake.Common/Tools/MSBuild/MSBuildRunner.cs
+++ b/src/Cake.Common/Tools/MSBuild/MSBuildRunner.cs
@@ -8,7 +8,6 @@ using System.Globalization;
 using System.Linq;
 using System.Text;
 using Cake.Core;
-using Cake.Core.Diagnostics;
 using Cake.Core.IO;
 using Cake.Core.Tooling;
 
@@ -72,7 +71,7 @@ namespace Cake.Common.Tools.MSBuild
             }
 
             // Set the verbosity.
-            builder.Append(string.Format(CultureInfo.InvariantCulture, "/v:{0}", GetVerbosityName(settings.Verbosity)));
+            builder.Append(string.Format(CultureInfo.InvariantCulture, "/v:{0}", settings.Verbosity.GetMSBuildVerbosityName()));
 
             if (settings.NodeReuse != null)
             {
@@ -196,24 +195,6 @@ namespace Cake.Common.Tools.MSBuild
                 default:
                     throw new ArgumentOutOfRangeException(nameof(platform), platform, "Invalid platform");
             }
-        }
-
-        private static string GetVerbosityName(Verbosity verbosity)
-        {
-            switch (verbosity)
-            {
-                case Verbosity.Quiet:
-                    return "quiet";
-                case Verbosity.Minimal:
-                    return "minimal";
-                case Verbosity.Normal:
-                    return "normal";
-                case Verbosity.Verbose:
-                    return "detailed";
-                case Verbosity.Diagnostic:
-                    return "diagnostic";
-            }
-            throw new CakeException("Encountered unknown MSBuild build log verbosity.");
         }
 
         private static IEnumerable<string> GetPropertyArguments(IDictionary<string, IList<string>> properties)

--- a/src/Cake.Common/Tools/MSBuild/MSBuildVerbosityExtensions.cs
+++ b/src/Cake.Common/Tools/MSBuild/MSBuildVerbosityExtensions.cs
@@ -1,0 +1,35 @@
+﻿using Cake.Core;
+using Cake.Core.Diagnostics;
+
+namespace Cake.Common.Tools.MSBuild
+{
+    /// <summary>
+    /// Contains functionality related to MSBuild verbosity.
+    /// </summary>
+    public static class MSBuildVerbosityExtensions
+    {
+        /// <summary>
+        /// Gets the MSBuild verbosity from <see cref="Verbosity"/>.
+        /// </summary>
+        /// <param name="verbosity">The verbosity.</param>
+        /// <returns>MSBuild verbosity string.</returns>
+        public static string GetMSBuildVerbosityName(this Verbosity verbosity)
+        {
+            switch (verbosity)
+            {
+                case Verbosity.Quiet:
+                    return "quiet";
+                case Verbosity.Minimal:
+                    return "minimal";
+                case Verbosity.Normal:
+                    return "normal";
+                case Verbosity.Verbose:
+                    return "detailed";
+                case Verbosity.Diagnostic:
+                    return "diagnostic";
+                default:
+                    throw new CakeException("Encountered unknown MSBuild build log verbosity.");
+            }
+        }
+    }
+}

